packages/reactivity/__tests__/computed.spec.ts

```diff
@@ -521,6 +522,36 @@ describe('reactivity/computed', () => {
     expect(fnSpy).toBeCalledTimes(2)
   })
 
+  // #10185
+  it('should not override queried MaybeDirty result', () => {
+    class Item {
+      v = ref(0)
+    }
+    const v1 = shallowRef()
+    const v2 = ref(false)
+    const c1 = computed(() => {
+      let c = v1.value
+      if (!v1.value) {
+        c = new Item()
+        v1.value = c
+      }
+      return c.v.value
+    })
+    const c2 = computed(() => {
+      if (!v2.value) return 'no'
+      return c1.value ? 'yes' : 'no'
+    })
+    const c3 = computed(() => c2.value)
+
+    c3.value
```

Comment:
Maybe we can add more dirtyLevel Assertions?
```typescript
    c3.value
    v2.value = true
    expect(c2.effect._dirtyLevel).toBe(DirtyLevels.Dirty)
    expect(c3.effect._dirtyLevel).toBe(DirtyLevels.MaybeDirty)

    c3.value
    expect(c1.effect._dirtyLevel).toBe(DirtyLevels.Dirty)
    expect(c2.effect._dirtyLevel).toBe(DirtyLevels.MaybeDirty)
    expect(c3.effect._dirtyLevel).toBe(DirtyLevels.MaybeDirty)

    v1.value.v.value = 999
    expect(c1.effect._dirtyLevel).toBe(DirtyLevels.Dirty)
    expect(c2.effect._dirtyLevel).toBe(DirtyLevels.MaybeDirty)
    expect(c3.effect._dirtyLevel).toBe(DirtyLevels.MaybeDirty)

    expect(c3.value).toBe('yes')
```

Reply:
This may make the test cases more instructive

Reply:
Done.